sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -45,8 +45,8 @@ case class DataSourceResolution(
   import org.apache.spark.sql.catalog.v2.CatalogV2Implicits._
   import lookup._
 
-  lazy val v2SessionCatalog: CatalogPlugin = lookup.sessionCatalog
-      .getOrElse(throw new AnalysisException("No v2 session catalog implementation is available"))
+  def v2SessionCatalog: CatalogPlugin = lookup.sessionCatalog
```

Comment:
Do we even need `LookupCatalog` anymore?

Reply:
Shouldn't this rule just get the `CatalogManager` as an input instead of `LookupCatalog`?

Reply:
The `LookupCatalog` has some convenient utils, e.g. `CatalogObjectIdentifier`, `AsTableIdentifier`, etc. I think we should still keep it.

BTW good point about making this rule take `CatalogManager` directly. Will update it soon.